### ISSUE TITLE
Premature `stream.end()` in `search()`

### ISSUE THREAD

**Issue by leesei** [jwerle/rfc#12]
[`parse()`](https://github.com/jwerle/rfc/blob/master/index.js#L189) calls `stream.end()` after looping through `RFC_CACHE_INDEX`, causing the whole pipeline to stop.
However, [`write()`](https://github.com/jwerle/rfc/blob/master/index.js#L238-L242) maybe calling `rfc.sync()` asynchronously.
One with good network connection may not encounter this for search that returns a few results.

Try:
```sh
$ rfc search protocol
  ... searching


  ... (0) result
# halts
```

- 'end' is signalled prematurely so `bin/rfc` shows 0 results
- app halts as `rfc.sync()`'s 'end' handler emit 'result' on an already ended stream

I have a fixed branch [here](https://github.com/leesei/rfc/commit/c24ffb86e76635b9c4ea2acd5c38a0adce66789d) where:
- search() will not have the side effect of sync-ing the RFC documents
- `RFC.sync()` must be called explicitly before `RFC.open()`

`bin/rfc`'s open is modified accordingly.
Please comment. I'll clean up and do separate commit.

**Comment by jwerle** [comment]
Nice catch. I've noticed this issue in the past. Feel free to issue a PR when it is ready to be merged

**Comment by leesei** [comment]
'K, PR coming.

Rationales for removing `RFC.sync()` from `search()`:
- reduce side effects of function call
- async `RFC.sync()` ruins the order of `search()` results
- premature 'end'

**Comment by leesei** [comment]
Fixed by #13
